lib/bulk_processor/job.rb

```diff
@@ -0,0 +1,28 @@
+class BulkProcessor
+  class Job < ActiveJob::Base
+    queue_as 'bulk_processor'
+
+    def perform(records, item_proccessor, handler, payload)
+      item_proccessor_class = item_proccessor.constantize
+      handler_class = handler.constantize
+
+      successes = {}
+      failures = {}
```

Comment:
I suspect the endgame of these is a `BulkProcessor::Row` that knows its index, data, messages, etc.